app/models/Language.py

```diff
@@ -0,0 +1,78 @@
+from misc import lang_path
+from json import loads as json_loads
+from hashlib import md5
+
+# JSON of languages
+with open(lang_path) as lang_data:
+    languages = json_loads(lang_data.read())
+
+class Language(object):
+    """
+    Supplies all known information about a language. Make an instance using the
+    language id. If a language does not exist then this will *not* error however
+    some functions will return `None` appropriately.
+    """
+    def __init__(self, lang_id):
+        self._id = Language.normalize(lang_id)
+
+        # Should not fail otherwise you have malformed source code
+        self._data = languages['languages'][self._id]
+
+    def get_color(self):
+        return '#' + "".join(md5(self._id.encode('ascii')).hexdigest()[:6])
```

Comment:
This will lead to too-similar colors, which will negatively impact accessibility by colorblind users. Use one of the palettes that I sent you.